### PR TITLE
Fixed #30812 -- Made ConditionalGetMiddleware set ETag only for responses with non-empty content.

### PR DESCRIPTION
**Pull request by Dart-net**

### CODE REVIEW

**Comment by claudep** [comment]
Please fix the failure and add tests!

File: django/utils/cache.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def get_max_age(response):
 
 
 def set_response_etag(response):
-    if not response.streaming:
+    if response.getvalue() and not response.streaming:
```
Comment (claudep) [inline django/utils/cache.py]: I think `response.streaming` should be tested first which will avoid accessing content for such responses.

Comment (charettes) [reply django/utils/cache.py]: Also what's the reason for using `getvalue()` over `.content`?

```suggestion
    if not response.streaming and response.content:
```

**Comment by Dart-net** [comment]
Thanks for help!

**Comment by felixxm** [comment]
@Dart-net Thanks :+1: I added a small note to backward incompatible changes.

**Pull request merged**